small fixes for rtjack compilation on win32

hlolli opened the pull request:
After long battle with jack2 on windows, I was able to make it run, and to be able to compile it, I needed to make minor changes to cmake and rtjack.c. But it wont run from the official jack2 installation, since it hasn't been updated since 2016. If someone really wants to use jack2 with windows and csound, I would recommend extracting the pre-built jack2 from http://users.notam02.no/~kjetism/radium/demos/windows64/?C=M;O=D there they distribute newer binaries with the same ABI. It's just a question of time when the jack developers make a new windows release and the errors from their current outdated release are fixed.

stekyne commented:
It seems VCPKG has v1.9.12, which is December 2017. Does that help much more?

hlolli commented:
Yes, we could recommend users to use VCPKG distribution of jack instead of the Radium one.

hlolli commented:
@stekyne I gave vcpkg a try, and the jack2 package they have there https://github.com/microsoft/vcpkg/tree/master/ports/jack2 is a x86 package missing most if not all of libjack. So it wont be of much use at this moment.

stekyne commented:
It installs x86 by default. You have to specify x64, so "vcpkg install jack2:x64-windows". 

What part is missing exactly? You may still need Jack to be installed on the client machine.

hlolli commented:
I can try that, but I've had great success with Radium bundles so far. But with only 1 .lib file from vckpg, it seems to miss libjack64.dll and libjackserver64.dll (likned by their .lib counterparts). That is still a seperate issue from this PR which is just about fixing the build of rtjack.dll. But since I haven't been able to get it to run, we can discuss a bit before merge-ing. Now that I've been able to make sure that rtjack.dll doesn't segfault and the initial dl_load of the libjackserver64.dll doesn't segfault either, I'm left with a new runtime bug. I'll post the debugger stacktrace here below

```
Microsoft (R) Windows Debugger Version 10.0.18362.1 AMD64
Copyright (c) Microsoft Corporation. All rights reserved.

CommandLine: .\csound.exe -+rtaudio=jack .\test.csd -odac -+jack_client=test
Symbol search path is: srv*
Executable search path is:
ModLoad: 00007ff7`50d20000 00007ff7`50d34000   csound.exe
ModLoad: 00007ffa`81520000 00007ffa`81700000   ntdll.dll
ModLoad: 00007ffa`7ef00000 00007ffa`7efae000   C:\Windows\System32\KERNEL32.DLL
ModLoad: 00007ffa`7df20000 00007ffa`7e186000   C:\Windows\System32\KERNELBASE.dll
ModLoad: 00007ffa`6c5b0000 00007ffa`6c5d1000   C:\Windows\SYSTEM32\VCRUNTIME140D.dll
ModLoad: 00007ffa`3e4f0000 00007ffa`3e98e000   C:\Users\User\Documents\csound\build\Debug\csound64.dll
ModLoad: 00007ffa`7de20000 00007ffa`7df16000   C:\Windows\System32\ucrtbase.dll
ModLoad: 00007ffa`80d60000 00007ffa`80dcc000   C:\Windows\System32\WS2_32.dll
ModLoad: 00007ffa`81370000 00007ffa`8148f000   C:\Windows\System32\RPCRT4.dll
ModLoad: 00007ffa`43740000 00007ffa`43903000   C:\Windows\SYSTEM32\ucrtbased.dll
ModLoad: 000001e4`04e50000 000001e4`05013000   C:\Windows\SYSTEM32\ucrtbased.dll
ModLoad: 00007ffa`6f3c0000 00007ffa`6f3d6000   C:\Windows\SYSTEM32\VCRUNTIME140.dll
ModLoad: 00000000`5ee20000 00000000`5efd6000   C:\Users\User\Documents\csound\build\Debug\libsndfile-1.dll
ModLoad: 00007ffa`7f050000 00007ffa`7f0ed000   C:\Windows\System32\msvcrt.dll
ModLoad: 00007ffa`81090000 00007ffa`8121f000   C:\Windows\System32\USER32.dll
ModLoad: 00007ffa`7dce0000 00007ffa`7dd00000   C:\Windows\System32\win32u.dll
ModLoad: 00007ffa`7ec00000 00007ffa`7ec28000   C:\Windows\System32\GDI32.dll
ModLoad: 00007ffa`7db40000 00007ffa`7dcd3000   C:\Windows\System32\gdi32full.dll
ModLoad: 00007ffa`7dd80000 00007ffa`7de1b000   C:\Windows\System32\msvcp_win.dll
(1bf8.1628): Break instruction exception - code 80000003 (first chance)
ntdll!LdrpDoDebuggerBreak+0x30:
00007ffa`815f2efc cc              int     3
0:000> g
ModLoad: 00007ffa`80dd0000 00007ffa`80dfd000   C:\Windows\System32\IMM32.DLL
ModLoad: 00007ffa`80a50000 00007ffa`80d58000   C:\Windows\System32\combase.dll
ModLoad: 00007ffa`7dd00000 00007ffa`7dd72000   C:\Windows\System32\bcryptPrimitives.dll
time resolution is 364.672 ns
ModLoad: 00007ffa`75aa0000 00007ffa`75ab3000   C:\Users\User\Documents\csound\build\Debug\ampmidid.dll
ModLoad: 00007ffa`70890000 00007ffa`708c3000   C:\Users\User\Documents\csound\build\Debug\arrayops.dll
ModLoad: 00007ffa`707a0000 00007ffa`70890000   C:\Windows\SYSTEM32\MSVCP140D.dll
ModLoad: 00007ffa`70780000 00007ffa`7079a000   C:\Users\User\Documents\csound\build\Debug\beosc.dll
ModLoad: 00007ffa`6fc00000 00007ffa`6fc14000   C:\Users\User\Documents\csound\build\Debug\buchla.dll
ModLoad: 00007ffa`6fbe0000 00007ffa`6fbf2000   C:\Users\User\Documents\csound\build\Debug\cellular.dll
ModLoad: 00007ffa`6f680000 00007ffa`6f711000   C:\Users\User\Documents\csound\build\Debug\csnd6.dll
ModLoad: 00007ffa`6fbc0000 00007ffa`6fbd3000   C:\Users\User\Documents\csound\build\Debug\cs_date.dll
ModLoad: 00007ffa`6fb90000 00007ffa`6fbb1000   C:\Users\User\Documents\csound\build\Debug\doppler.dll
ModLoad: 00007ffa`6fb70000 00007ffa`6fb8c000   C:\Users\User\Documents\csound\build\Debug\emugens.dll
ModLoad: 00007ffa`6f700000 00007ffa`6f714000   C:\Users\User\Documents\csound\build\Debug\exciter.dll
ModLoad: 00007ffa`6f6e0000 00007ffa`6f6f3000   C:\Users\User\Documents\csound\build\Debug\fareygen.dll
ModLoad: 00007ffa`6f6c0000 00007ffa`6f6d6000   C:\Users\User\Documents\csound\build\Debug\fractalnoise.dll
ModLoad: 00007ffa`6f6a0000 00007ffa`6f6b3000   C:\Users\User\Documents\csound\build\Debug\framebuffer.dll
ModLoad: 00007ffa`6f670000 00007ffa`6f699000   C:\Users\User\Documents\csound\build\Debug\ftsamplebank.dll
ModLoad: 00007ffa`6f650000 00007ffa`6f662000   C:\Users\User\Documents\csound\build\Debug\getftargs.dll
ModLoad: 00007ffa`6f630000 00007ffa`6f642000   C:\Users\User\Documents\csound\build\Debug\gtf.dll
ModLoad: 00007ffa`6f610000 00007ffa`6f623000   C:\Users\User\Documents\csound\build\Debug\ipmidi.dll
ModLoad: 00000000`61440000 00000000`6155e000   C:\Users\User\Documents\csound\build\Debug\libgcc_s_seh-1.dll
ModLoad: 00000000`71040000 00000000`710dd000   C:\Users\User\Documents\csound\build\Debug\libgnurx-0.dll
ModLoad: 00000000`69bc0000 00000000`69c6b000   C:\Users\User\Documents\csound\build\Debug\libjack64.dll
ModLoad: 00007ffa`7ecb0000 00007ffa`7ed51000   C:\Windows\System32\ADVAPI32.dll
ModLoad: 00007ffa`7eb60000 00007ffa`7ebbb000   C:\Windows\System32\sechost.dll
ModLoad: 00007ffa`80f10000 00007ffa`80f18000   C:\Windows\System32\PSAPI.DLL
ModLoad: 00007ffa`7f540000 00007ffa`80976000   C:\Windows\System32\SHELL32.dll
ModLoad: 00007ffa`7d920000 00007ffa`7d96a000   C:\Windows\System32\cfgmgr32.dll
ModLoad: 00007ffa`80e00000 00007ffa`80ea6000   C:\Windows\System32\shcore.dll
ModLoad: 00007ffa`7e190000 00007ffa`7e8d7000   C:\Windows\System32\windows.storage.dll
ModLoad: 00007ffa`80eb0000 00007ffa`80f01000   C:\Windows\System32\shlwapi.dll
ModLoad: 00007ffa`7d870000 00007ffa`7d881000   C:\Windows\System32\kernel.appcore.dll
ModLoad: 00007ffa`7d8b0000 00007ffa`7d8fc000   C:\Windows\System32\powrprof.dll
ModLoad: 00007ffa`7d890000 00007ffa`7d8ab000   C:\Windows\System32\profapi.dll
ModLoad: 00000000`71040000 00000000`710dd000   C:\Users\User\Documents\csound\build\Debug\libgnurx-0.dll
ModLoad: 00007ffa`7aa00000 00007ffa`7aa23000   C:\Windows\SYSTEM32\WINMM.dll
ModLoad: 00000000`61440000 00000000`6155e000   C:\Users\User\Documents\csound\build\Debug\libgcc_s_seh-1.dll
ModLoad: 00007ffa`7a9d0000 00007ffa`7a9fa000   C:\Windows\SYSTEM32\WINMMBASE.dll
ModLoad: 00000000`6fc40000 00000000`6fd9e000   C:\Users\User\Documents\csound\build\Debug\libstdc++-6.dll
ModLoad: 00000000`68440000 00000000`68541000   C:\Users\User\Documents\csound\build\Debug\libjackserver-0.dll
ModLoad: 00007ffa`7ecb0000 00007ffa`7ed51000   C:\Windows\System32\ADVAPI32.dll
ModLoad: 00007ffa`7eb60000 00007ffa`7ebbb000   C:\Windows\System32\sechost.dll
ModLoad: 00007ffa`80f10000 00007ffa`80f18000   C:\Windows\System32\PSAPI.DLL
ModLoad: 00000000`71040000 00000000`710dd000   C:\Users\User\Documents\csound\build\Debug\libgnurx-0.dll
ModLoad: 00007ffa`7aa00000 00007ffa`7aa23000   C:\Windows\SYSTEM32\WINMM.dll
ModLoad: 00000000`61440000 00000000`6155e000   C:\Users\User\Documents\csound\build\Debug\libgcc_s_seh-1.dll
ModLoad: 00000000`6fc40000 00000000`6fd9e000   C:\Users\User\Documents\csound\build\Debug\libstdc++-6.dll
ModLoad: 00007ffa`7a9d0000 00007ffa`7a9fa000   C:\Windows\SYSTEM32\WINMMBASE.dll
ModLoad: 00007ffa`7d920000 00007ffa`7d96a000   C:\Windows\System32\cfgmgr32.dll
ModLoad: 00000000`632c0000 00000000`63446000   C:\Users\User\Documents\csound\build\Debug\libportaudio-2.dll
ModLoad: 00007ffa`7ecb0000 00007ffa`7ed51000   C:\Windows\System32\ADVAPI32.dll
ModLoad: 00007ffa`7eb60000 00007ffa`7ebbb000   C:\Windows\System32\sechost.dll
ModLoad: 00007ffa`81220000 00007ffa`81369000   C:\Windows\System32\ole32.dll
ModLoad: 00007ffa`7f0f0000 00007ffa`7f53e000   C:\Windows\System32\SETUPAPI.dll
ModLoad: 00007ffa`7d920000 00007ffa`7d96a000   C:\Windows\System32\cfgmgr32.dll
ModLoad: 00000000`61440000 00000000`6155e000   C:\Users\User\Documents\csound\build\Debug\libgcc_s_seh-1.dll
ModLoad: 00000000`6fc40000 00000000`6fd9e000   C:\Users\User\Documents\csound\build\Debug\libstdc++-6.dll
ModLoad: 00007ffa`7aa00000 00007ffa`7aa23000   C:\Windows\SYSTEM32\WINMM.dll
ModLoad: 00007ffa`7a9d0000 00007ffa`7a9fa000   C:\Windows\SYSTEM32\WINMMBASE.dll
ModLoad: 00000000`6fc40000 00000000`6fd9e000   C:\Users\User\Documents\csound\build\Debug\libstdc++-6.dll
ModLoad: 00000000`61440000 00000000`6155e000   C:\Users\User\Documents\csound\build\Debug\libgcc_s_seh-1.dll
ModLoad: 00007ffa`6f5f0000 00007ffa`6f603000   C:\Users\User\Documents\csound\build\Debug\liveconv.dll
ModLoad: 00007ffa`6f590000 00007ffa`6f5ed000   C:\Users\User\Documents\csound\build\Debug\mixer.dll
ModLoad: 00007ffa`6cda0000 00007ffa`6cdb9000   C:\Users\User\Documents\csound\build\Debug\padsynth.dll
ModLoad: 00007ffa`6c520000 00007ffa`6c534000   C:\Users\User\Documents\csound\build\Debug\platerev.dll
ModLoad: 00007ffa`6c100000 00007ffa`6c123000   C:\Users\User\Documents\csound\build\Debug\pvsops.dll
ModLoad: 00007ffa`6c500000 00007ffa`6c512000   C:\Users\User\Documents\csound\build\Debug\quadbezier.dll
ModLoad: 00007ffa`6b010000 00007ffa`6b02c000   C:\Users\User\Documents\csound\build\Debug\rtjack.dll
ModLoad: 00000000`68440000 00000000`68541000   C:\Users\User\Documents\csound\build\Debug\libjackserver-0.dll
ModLoad: 00007ffa`7ecb0000 00007ffa`7ed51000   C:\Windows\System32\ADVAPI32.dll
ModLoad: 00007ffa`7eb60000 00007ffa`7ebbb000   C:\Windows\System32\sechost.dll
ModLoad: 00007ffa`80f10000 00007ffa`80f18000   C:\Windows\System32\PSAPI.DLL
ModLoad: 00000000`71040000 00000000`710dd000   C:\Users\User\Documents\csound\build\Debug\libgnurx-0.dll
ModLoad: 00007ffa`7aa00000 00007ffa`7aa23000   C:\Windows\SYSTEM32\WINMM.dll
ModLoad: 00000000`61440000 00000000`6155e000   C:\Users\User\Documents\csound\build\Debug\libgcc_s_seh-1.dll
ModLoad: 00000000`6fc40000 00000000`6fd9e000   C:\Users\User\Documents\csound\build\Debug\libstdc++-6.dll
ModLoad: 00007ffa`7a9d0000 00007ffa`7a9fa000   C:\Windows\SYSTEM32\WINMMBASE.dll
ModLoad: 00007ffa`7d920000 00007ffa`7d96a000   C:\Windows\System32\cfgmgr32.dll
ModLoad: 00007ffa`661e0000 00007ffa`661f7000   C:\Users\User\Documents\csound\build\Debug\rtwinmm.dll
ModLoad: 00007ffa`658f0000 00007ffa`6590c000   C:\Users\User\Documents\csound\build\Debug\scansyn.dll
ModLoad: 00007ffa`658d0000 00007ffa`658e4000   C:\Users\User\Documents\csound\build\Debug\scugens.dll
ModLoad: 00007ffa`63660000 00007ffa`63672000   C:\Users\User\Documents\csound\build\Debug\select.dll
ModLoad: 00007ffa`62ab0000 00007ffa`62ac3000   C:\Users\User\Documents\csound\build\Debug\serial.dll
ModLoad: 00007ffa`61d10000 00007ffa`61e01000   C:\Users\User\Documents\csound\build\Debug\signalflowgraph.dll
ModLoad: 00007ffa`62a90000 00007ffa`62aa4000   C:\Users\User\Documents\csound\build\Debug\stackops.dll
ModLoad: 00007ffa`62480000 00007ffa`624d9000   C:\Users\User\Documents\csound\build\Debug\stdutil.dll
ModLoad: 00007ffa`62920000 00007ffa`62932000   C:\Users\User\Documents\csound\build\Debug\system_call.dll
0dBFS level = 32768.0
--Csound version 6.13 beta (double samples) Jun 11 2019
[commit: daddfddc88a42bdba97021498c938932d0fcebb8]
libsndfile-1.0.28
UnifiedCSD:  .\test.csd
STARTING FILE
Creating options
0: csound
1: -odac
Creating orchestra
closing tag
Creating score
rtaudio: JACK module enabled
Elapsed time at end of orchestra compile: real: 0.017s, CPU: 0.016s
sorting score ...
        ... done
Elapsed time at end of score sort: real: 0.018s, CPU: 0.018s
graphics suppressed, ascii substituted
0dBFS level = 32768.0
orch now loaded
audio buffered in 4096 sample-frame blocks
Drivers/internals found in : C:\Users\User\Documents\csound\build\Debug
ModLoad: 00000000`6b800000 00000000`6b828000   C:\Users\User\Documents\csound\build\Debug\jack\jack_dummy.dll
ModLoad: 00000000`6b800000 00000000`6b828000   C:\Users\User\Documents\csound\build\Debug\jack\jack_dummy.dll
ModLoad: 00000000`6d040000 00000000`6d088000   C:\Users\User\Documents\csound\build\Debug\jack\jack_portaudio.dll
ModLoad: 00000000`632c0000 00000000`63446000   C:\Users\User\Documents\csound\build\Debug\libportaudio-2.dll
ModLoad: 00007ffa`81220000 00007ffa`81369000   C:\Windows\System32\ole32.dll
ModLoad: 00007ffa`7f0f0000 00007ffa`7f53e000   C:\Windows\System32\SETUPAPI.dll
ModLoad: 00000000`6d040000 00000000`6d088000   C:\Users\User\Documents\csound\build\Debug\jack\jack_portaudio.dll
ModLoad: 00000000`632c0000 00000000`63446000   C:\Users\User\Documents\csound\build\Debug\libportaudio-2.dll
ModLoad: 00007ffa`81220000 00007ffa`81369000   C:\Windows\System32\ole32.dll
ModLoad: 00007ffa`7f0f0000 00007ffa`7f53e000   C:\Windows\System32\SETUPAPI.dll
(1bf8.1628): Access violation - code c0000005 (first chance)
First chance exceptions are reported before any exception handling.
This exception may be expected and handled.
msvcrt!strcmp+0x43:
00007ffa`7f0abb13 4a8b1409        mov     rdx,qword ptr [rcx+r9] ds:00000000`00000000=????????????????
0:000> k
Child-SP          RetAddr           Call Site
000000b2`13d8ec58 00000000`684667f3 msvcrt!strcmp+0x43
000000b2`13d8ec60 00000000`684684e9 libjackserver_0!jackctl_driver_params_parse+0x8be
000000b2`13d8eca0 00000000`68464cec libjackserver_0!ZN4Jack17JackServerGlobals4InitEv+0x7f7
000000b2`13d8f440 00000000`68464e5e libjackserver_0!ZN4Jack22JackWaitCallbackDriver11ExecuteRealEv+0x3f6
*** WARNING: Unable to verify checksum for C:\Users\User\Documents\csound\build\Debug\rtjack.dll
000000b2`13d8f4e0 00007ffa`6b01374f libjackserver_0!jack_client_open+0x87
000000b2`13d8f540 00007ffa`6b014f11 rtjack!openJackStreams+0x6f
*** WARNING: Unable to verify checksum for C:\Users\User\Documents\csound\build\Debug\csound64.dll
000000b2`13d8f880 00007ffa`3e54d67b rtjack!playopen_+0x161
000000b2`13d8f8c0 00007ffa`3e53429f csound64!sfopenout+0x31b
000000b2`13d8fb30 00007ffa`3e532ff1 csound64!csoundInitialiseIO+0x2bf
000000b2`13d8fb90 00007ffa`3e727d77 csound64!musmon+0x5c1
000000b2`13d8fc10 00007ffa`3e727ddf csound64!csoundStart+0x6d7
*** WARNING: Unable to verify checksum for csound.exe
000000b2`13d8fc80 00007ff7`50d21a74 csound64!csoundCompile+0x4f
000000b2`13d8fcc0 00007ff7`50d224c4 csound!main+0x254
000000b2`13d8fd20 00007ff7`50d22377 csound!invoke_main+0x34
000000b2`13d8fd60 00007ff7`50d2223e csound!__scrt_common_main_seh+0x127
000000b2`13d8fdc0 00007ff7`50d22559 csound!__scrt_common_main+0xe
000000b2`13d8fdf0 00007ffa`7ef11fe4 csound!mainCRTStartup+0x9
000000b2`13d8fe20 00007ffa`8158efc1 KERNEL32!BaseThreadInitThunk+0x14
000000b2`13d8fe50 00000000`00000000 ntdll!RtlUserThreadStart+0x21
0:000>
```

It seems to be failing around here https://github.com/jackaudio/jack2/blob/98c882c5a545c3dd819655c9869f6928274df8a3/common/JackDriverLoader.cpp#L286-L289

my initial reaction was that we shouldn't be using jackserver(jackdmp) but libjack64 only. So I compiled rtjack that way, but then I got stubled on jack functions being called that should only exists in the server api. Maybe csound needs to provide some jack master info into the Jack struct? Since jack on windows is kinda a piggybagging over other audio hosts (ASIO/Portaudio)

kunstmusik commented:
@stekyne @hlolli Following up with this, does this need further changes or can it be merged? I don't use JACK on Windows so I can not comment on this, but we're looking to sort out issues/PRs for the next release.

hlolli commented:
This needs change, sorry, I'll try to get it done soon.

On Sat, 6 Jul 2019 at 22:15, Steven Yi <notifications@github.com> wrote:

> @stekyne <https://github.com/stekyne> @hlolli <https://github.com/hlolli>
> Following up with this, does this need further changes or can it be merged?
> I don't use JACK on Windows so I can not comment on this, but we're looking
> to sort out issues/PRs for the next release.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/csound/csound/pull/1151?email_source=notifications&email_token=ABOLDAROORGRF6BKH5ROGBLP6D4NDA5CNFSM4HWZLTLKYY3PNVWWK3TUL52HS4DFVREXG43VMVBW63LNMVXHJKTDN5WW2ZLOORPWSZGODZK7T4I#issuecomment-508951025>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ABOLDASC2C2CB625KLU2NXLP6D4NDANCNFSM4HWZLTLA>
> .
>

hlolli commented:
I've cleaned this commit, so I did 2 things

1. I removed any reference to JACKDMP, this was what was bugging me the most last summer, because trying to link against the jack-server (a front-end) gave my vague and strange bugs. This may have worked in the past since jack-server/jackd/jackdmp exports most of the same symbols as jack, but this just wont work, and csound doesn't need to depend at all on jack-server/jackd/jackdmp (even tough it's compiled along with jack).

2. jack/jack.h wont get included, I saw that this was a problem on Mac as well (where VL wrote a workaround), but when I include an include directory (in jack2's case jack/common), it will find jack.h. It's stored as JACK_INCLUDE_DIRECTORY and it would be a required flag for all libjack compilation on win32.

jpffitch commented:
Seems to fail on appveyor with missing dependencies such as vorbis

⁣Sent from TypeApp ​

On Oct 15, 2019, 03:13, at 03:13, "Hlöðver Sigurðsson" <notifications@github.com> wrote:
>I've cleaned this commit, so I did 2 things
>
>1. I removed any reference to JACKDMP, this was what was bugging me the
>most last summer, because trying to link against the jack-server (a
>front-end) gave my vague and strange bugs. This may have worked in the
>past since jack-server/jackd/jackdmp exports most of the same symbols
>as jack, but this just wont work, and csound doesn't need to depend at
>all on jack-server/jackd/jackdmp (even tough it's compiled along with
>jack).
>
>2. jack/jack.h wont get included, I saw that this was a problem on Mac
>as well (where VL wrote a workaround), but when I include an include
>directory (in jack2's case jack/common), it will find jack.h. It's
>stored as JACK_INCLUDE_DIRECTORY and it would be a required flag for
>all libjack compilation on win32.
>
>
>-- 
>You are receiving this because you are subscribed to this thread.
>Reply to this email directly or view it on GitHub:
>https://github.com/csound/csound/pull/1151#issuecomment-542004541

hlolli commented:
That would be #1192 that caused it @jpffitch

kunstmusik commented:
This looks good to me; as noted in another issue, we need to audit our CMake system to do searches for includes and add the paths to the targets like this change does.  

Tests have passed, I am merging now. Thanks @hlolli !
